pkg/controller/core/cohort_controller.go

```diff
@@ -110,5 +150,71 @@ func (r *CohortReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		log.V(2).Error(err, "Error adding or updating cohort in the cache")
 	}
 	r.qManager.AddOrUpdateCohort(ctx, &cohort)
-	return ctrl.Result{}, nil
+
+	err := r.updateCohortStatusIfChanged(ctx, &cohort)
+	return ctrl.Result{}, err
+}
+
+func (r *CohortReconciler) updateCohortStatusIfChanged(ctx context.Context, cohort *kueue.Cohort) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	oldStatus := cohort.Status.DeepCopy()
+
+	stats, err := r.cache.CohortStats(cohort)
+	if err != nil {
+		log.Error(err, "Failed getting cohort usage from cache")
+		// This is likely because the cohort was recently removed,
+		// but we didn't process that event yet.
```

Comment:
I don't think this is accurate, as we will only delete Cohort in Reconcile loop, which we are currently in